server/src/main/java/com/cloud/resourcelimit/ResourceLimitManagerImpl.java

```diff
@@ -1171,6 +1178,27 @@ public ResourceCountCheckTask() {
 
         @Override
         protected void runInContext() {
+            GlobalLock lock = GlobalLock.getInternLock("ResourceCheckTask");
+            try {
+                if (lock.lock(30)) {
+                    ManagementServerHostVO msHost = managementServerHostDao.findOneByLongestRuntime();
+                    if (msHost == null || (msHost.getMsid() != ManagementServerNode.getManagementServerId())) {
+                        s_logger.debug("Skipping the resource counters recalculation task on this management server");
+                        lock.unlock();
+                        return;
```

Comment:
does this mean the `lock.releaseRef()` is not needed?

Reply:
No, the releaseRef will be executed after that as its part of the finally block